From 44cafc3d5aa227d2afb174b36c2310c8492a9ac7 Mon Sep 17 00:00:00 2001
From: Prerna Kalla <prernak@codeaurora.org>
Date: Tue, 19 Nov 2019 12:11:48 +0530
Subject: [PATCH 3/3] LockSettingsService : Restrict access to getpassword API

Restrict Permission of getpassword API to ACCESS_KEYGUARD_SECURE_
STORAGE to ensure saftey for user credentials. No process without
this permission should be able to invoke this API from locksetting
aidl.

CRs-Fixed: 2576302
Change-Id: I7085a00acbdb3e0ea246210207e83c80ab48dc38
Signed-off-by: Prerna Kalla <prernak@codeaurora.org>
---
 .../com/android/server/locksettings/LockSettingsService.java    | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/services/core/java/com/android/server/locksettings/LockSettingsService.java b/services/core/java/com/android/server/locksettings/LockSettingsService.java
index 599cd7e2390f..272abbab2a23 100644
--- a/services/core/java/com/android/server/locksettings/LockSettingsService.java
+++ b/services/core/java/com/android/server/locksettings/LockSettingsService.java
@@ -1344,7 +1344,7 @@ public class LockSettingsService extends ILockSettings.Stub {
          */
        if (checkCryptKeeperPermissions())
             mContext.enforceCallingOrSelfPermission(
-                    android.Manifest.permission.MANAGE_DEVICE_ADMINS,
+                    android.Manifest.permission.ACCESS_KEYGUARD_SECURE_STORAGE,
                     "no crypt_keeper or admin permission to get the password");
 
        return mSavePassword;
-- 
2.30.2

